tests/test_random_order.py

```diff
@@ -98,5 +104,37 @@ def test_inverse(self, transform, invertible, use_metatensor):
                 self.assertDictEqual(fwd_data[i], _fwd_inv_data)
 
 
+TEST_RANDOM_ORDER_EXTENDED_TEST_CASES = [
+    [None, tuple()],
+    [None, (sa.Rotate(np.pi / 8),)],
+    [None, (sa.Flip(0), sa.Flip(1), sa.Rotate90(1), sa.Zoom(0.8), ia.NormalizeIntensity())],
+    [("a",), (sd.Rotated(("a",), np.pi / 8),)],
+]
+
+
+class TestRandomOrderAPITests(unittest.TestCase):
+    @staticmethod
+    def data_from_keys(keys):
+        if keys is None:
+            data = torch.unsqueeze(torch.tensor(np.arange(12 * 16).reshape(12, 16)), dim=0)
+        else:
+            data = {}
+            for i_k, k in enumerate(keys):
+                data[k] = torch.unsqueeze(torch.tensor(np.arange(12 * 16)).reshape(12, 16) + i_k * 192, dim=0)
+        return data
+
+    @parameterized.expand(TEST_RANDOM_ORDER_EXTENDED_TEST_CASES)
+    def test_execute_change_start_end(self, keys, pipeline):
+        data = self.data_from_keys(keys)
+
+        with self.assertRaises(ValueError):
+            c = RandomOrder(deepcopy(pipeline))
+            c(data, start=1)
+
+        with self.assertRaises(ValueError):
+            c = RandomOrder(deepcopy(pipeline))
+            c(data, end=1)
```

Comment:
```suggestion
        c = RandomOrder(deepcopy(pipeline))
        with self.assertRaises(ValueError):
            c(data, start=1)

        with self.assertRaises(ValueError):
            c(data, end=1)
```

Reply:
Done slightly differently to the suggestion but following the advice